internal/ctl/app/version.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/yaml"
+
+	"github.com/envoyproxy/gateway/internal/cmd/version"
+	"github.com/envoyproxy/gateway/internal/ctl/options"
+	"github.com/envoyproxy/gateway/internal/utils/kube"
+)
+
+func NewVersionsCommand() *cobra.Command {
+	var (
+		output string
+		remote bool
+	)
+
+	versionCommand := &cobra.Command{
+		Use:     "versions",
+		Aliases: []string{"version"},
+		Short:   "Show versions",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return versions(cmd.OutOrStdout(), remote, output)
+		},
+	}
+
+	flags := versionCommand.Flags()
+	options.AddKubeConfigFlags(flags)
+	versionCommand.PersistentFlags().BoolVar(&remote, "remote", false, "if true, retrieve server version from cluster")
+	versionCommand.PersistentFlags().StringVarP(&output, "output", "o", "json", "One of 'yaml' or 'json'")
```

Comment:
Since this is user-facing, I prefer yaml as the default over json.

```
$ egctl version --remote -o yaml
client: v0.2.0
servers:
  envoy-gateway-system/envoy-gateway-84f8b94f-8tqq7:
    envoyGatewayVersion: v0.2.0
    envoyVersion: latest
    gatewayAPIVersion: v0.6.0
    gitCommitID: 6f8b6fbeab204f9d5b334fb1e1cb6fbdc8b33112
```

Can you also update the following to the version output:
- Have the version command return `server` instead of `servers.
- Update `envoyVersion` to `envoyProxyVersion`.
- Parse `envoy-gateway-system/envoy-gateway-84f8b94f-8tqq7` to provide additional context in the output. For example:
```
namespace: envoy-gateway-system
deploymentName: envoy-gateway
```